### ISSUE TITLE
Unsupported components can be added to prefabs, undoing this action may result in Editor crash

### ISSUE THREAD

**Issue by LB-JakubSkorupka** [o3de/o3de#5649]
**Describe the bug**
It is possible to add components that are not supported by prefabs (by default only **Comment** and **AssetCollectionAsyncLoaderTest** should be available) to a prefab in the **Entity Inspector**. Undoing that action may result in one of the following outcomes:

1. Editor crashes.
2. Added component is removed from the prefab. In the **Entity Outliner** an expand/collapse arrow is displayed on the left side of the prefab, hovering the mouse cursor over the prefab will change that arrow back into the Focus Mode button (It is possible to return the prefab to its default state by pressing **Esc** or double-clicking the prefab in the **Entity Outliner** and entering Focus Mode again).
3. Added component is removed from the prefab, no other changes occur.

**Steps to reproduce**
1. Create a new Level.
2. Create a prefab.
3. Double click on prefab to enter Focus Mode.
4. Select the prefab (not the entity contained in the prefab).
5. Add a component that is not supported by prefabs to the prefab (if only **Comment** and **AssetCollectionAsyncLoaderTest** Components are be available, select the entity in the prefab and press "Add Component..." in the **Entity Inspector**, then attempt to add a component to the prefab itself again.
6. Undo the action (Ctrl + Z).

**Expected behavior**
Unsupported components cannot be used in prefabs.

**Actual behavior**
Unsupported components can be added to prefabs, which may result in an Editor crash.

**Call Stack Trace:**
```
25) 00007FFD35AAB7F7 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFD381151B0 (ntdll) : memset
23) 00007FFD380FC766 (ntdll) : _C_specific_handler
22) 00007FFD381120CF (ntdll) : _chkstk
21) 00007FFD380C1454 (ntdll) : RtlRaiseException
20) 00007FFD38110BFE (ntdll) : KiUserExceptionDispatcher
19) C:\builds\o3de\Code\Framework\AzCore\AzCore\IO\Path\PathParser.inl (375) : AZ::IO::parser::PathParser::Decrement
18) C:\builds\o3de\Code\Framework\AzCore\AzCore\IO\Path\Path.inl (211) : AZ::IO::PathView::filename_view
17) C:\builds\o3de\Code\Framework\AzCore\AzCore\IO\Path\Path.inl (1161) : AZ::IO::BasicPath<AZStd::basic_string<char,AZStd::char_traits<char>,AZStd::allocator> >::Filename
16) C:\builds\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Prefab\PrefabFocusHandler.cpp (372) : AzToolsFramework::Prefab::PrefabFocusHandler::RefreshInstanceFocusPath
15) C:\builds\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Prefab\PrefabFocusHandler.cpp (311) : AzToolsFramework::Prefab::PrefabFocusHandler::OnPrefabInstancePropagationEnd
14) C:\builds\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1360) : AZ::Internal::EBusContainer<AzToolsFramework::Prefab::PrefabPublicNotifications,AzToolsFramework::Prefab::PrefabPublicNotifications,0,1>::Dispatcher<AZ::EBus<AzToolsFramewo
13) C:\builds\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Prefab\Instance\InstanceUpdateExecutor.cpp (251) : AzToolsFramework::Prefab::InstanceUpdateExecutor::UpdateTemplateInstancesInQueue
12) C:\builds\o3de\Code\Framework\AzCore\AzCore\EBus\Internal\BusContainer.h (1365) : AZ::Internal::EBusContainer<AZ::SystemTickEvents,AZ::SystemTickEvents,0,1>::Dispatcher<AZ::EBus<AZ::SystemTickEvents,AZ::SystemTickEvents> >::Broadcast<void (__cdecl AZ::Sy
11) C:\builds\o3de\Code\Framework\AzCore\AzCore\Component\ComponentApplication.cpp (1449) : AZ::ComponentApplication::TickSystem
10) C:\builds\o3de\Code\Editor\CryEdit.cpp (2379) : CCryEditApp::IdleProcessing
 9) C:\builds\o3de\Code\Editor\CryEdit.cpp (2265) : CCryEditApp::OnIdle
 8) 00007FFCFD6F0DB9 (Qt5Core) : QObject::qt_static_metacall
 7) 00007FFCFD6FFA8D (Qt5Core) : QTimer::timerEvent
 6) 00007FFCFD6F854D (Qt5Core) : QObject::event
 5) 00007FFCFE1649B0 (Qt5Widgets) : QApplicationPrivate::notify_helper
 4) 00007FFCFE163A33 (Qt5Widgets) : QApplication::notify
 3) 00007FFCFD6D1B4A (Qt5Core) : QCoreApplication::notifyInternal2
 2) 00007FFCFD71C439 (Qt5Core) : QEventDispatcherWin32Private::sendTimerEvent
 1) 00007FFCFD71A869 (Qt5Core) : QEventDispatcherWin32::event
```

**Video**

https://user-images.githubusercontent.com/86951967/142005737-df9b12de-154c-403a-859c-7dd492e023af.mp4

**Found in Branch**
Stabilization_2110 (7031147)

**Desktop:**
 - Device: PC
 - OS: Windows
 - Version: 10
 - CPU: Intel Core i7-8700K
 - GPU: NVIDIA GeForce RTX 2080 Ti
 - Memory: 16GB

**Additional context**
[error.log](https://github.com/o3de/o3de/files/7547315/error.log)
[error.zip](https://github.com/o3de/o3de/files/7547316/error.zip)
[Editor.log](https://github.com/o3de/o3de/files/7547319/Editor.log)

**Comment by AMZN-daimini** [comment]
Due to a change introduced by this fix: https://github.com/o3de/o3de/pull/5747
I have verified the repro steps above no longer result in a crash.

On whether you should be able to add components to a container entities, I will raise that as a UX discussion but that's beyond the scope of this issue :)

**Comment by LB-JakubSkorupka** [comment]
Fix verified on Stabilization_2110 (a0ab592), reproducing the issue does not result in Editor crash.
